src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -301,6 +312,45 @@ public void RunTests(TestRunRequestPayload testRunRequestPayload, ITestHostLaunc
             }
         }
 
+        /// <inheritdoc/>
+        public void FinalizeMultiTestRun(MultiTestRunFinalizationPayload finalizationPayload, IMultiTestRunFinalizationEventsHandler finalizationEventsHandler, ProtocolConfig protocolConfig)
```

Comment:
Perhaps a similar natured question was asked in the RFC as well, but any reason this method is not async?

Reply:
You wouldn't need a separate Cancel API below in that case.

---
In reply to: [448174262](https://github.com/microsoft/vstest/pull/2463#discussion_r448174262) [](ancestors = 448174262)

Reply:
In RFC it was different place in code which was already fixed. Here we cant use async method as we can't use await inside lock. It also would not help with canceling as this would mean that DesignClient would need to keep cancellationtoken. I would be consistent here.